File: 1-Introductory_Problems/11-Coin_Piles/solution.cpp
#include <iostream>

int	main(void)
{
	long long	a;
	long long	b;
	int			t;

	std::cin >> t;
	while (t--)
	{
		std::cin >> a >> b;
		if ((a + b) % 3 || abs(a - b) > std::min(a, b))
			std::cout << "NO\n";
		else
			std::cout << "YES\n";
	}
	return (0);
}

File: 1-Introductory_Problems/11-Coin_Piles/solution.hs
import Control.Monad (forM_)

main :: IO ()
main = do
  n <- read <$> getLine :: IO Integer
  forM_ [1.. n] checkPile
    where checkPile _ = do
              [a, b] <- map read . words <$> getLine :: IO [Integer]
              if mod (a + b) 3 /= 0 || abs (a - b) > min a b
                then putStrLn "NO"
                else putStrLn "YES"